nestkernel/exceptions.cpp

```diff
@@ -408,3 +408,12 @@ nest::NumericalInstability::message() const
       << "updating " << model_ << ".";
   return msg.str();
 }
+
+std::string
+nest::NoEntryToMap::message() const
+{
+  std::ostringstream msg;
+  msg << "DynamicRecordablesMap::erase( const Name& n ): "
```

Comment:
With this text, this exception can only be thrown from `DynamicRecordablesMap::erase()`. It is usually not a good idea to constrain exceptions in this way. For greater flexibility, the function throwing the exception can pass its name to the exception.

Reply:
Changed to KeyError and made the error message more generic in eb41277